File: Core/Models/VariableGroup.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NXS.Core.Models
{
    [Table("VariableGroups")]
    public class VariableGroup
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; }
        public ICollection<Variable> Variables { get; set; }

        public VariableGroup() {
            Variables = new Collection<Variable>();
        }
    }
}

File: ClientApp/app/components/services/graph/variableGroup.ts

import { Variable } from "./variable";

export class VariableGroup {
    id: string;
    name: string;
    variables: Variable[]

    constructor(idValue: string, nameValue: string, variables: Variable[]) {
        this.id = idValue;
        this.name = nameValue;
        this.variables = variables;
    }
}